testing/hardhat-deployment/dependency/dependency.sh

```
@@ -0,0 +1,25 @@
+# LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+# LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+#
+# TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+# AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+# EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+# TITLE.
+
+set -e
+apk update && apk add --no-cache python3 make g++ build-base || { echo "Failed to install system packages"; exit 1; }
+cd /app/contracts && npm install || { echo "Failed to install Node.js dependencies"; exit 1; }
```

Shellcheck

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)